### DISCUSSION TITLE
Feature request: select file when opening directory

### DISCUSSION THREAD

**Discussion by pglpm** [cboxdoerfer/fsearch#440] [Ideas]
Fsearch (0.3.alpha0) has the great possibility of opening the directory containing an item in the result list, if one clicks on the "path" column. This is useful if one has to do extra file operations with the file from the result list, or maybe compare files that have close modification times (by time-sorting the directory). In my case (Ubuntu 20.04, KDE 5.18.8) the directory is simply opened, and I need to look for the file there, which can take a while with large directories, and sometimes requires starting a local search (with Dolphin in my case).

If possible it would be convenient to have the file automatically selected within the directory, when the latter is opened by Fsearch; possibly as an option in the settings. Another possibility is to allow the specification of a program to perform the directory-opening. For example, if I call `dolphin [filename]`, then the file directory is opened and the file is also selected therein.

Cheers!

**Comment by cboxdoerfer** [comment]
> If possible it would be convenient to have the file automatically selected within the directory, when the latter is opened by Fsearch; possibly as an option in the settings.

Unfortunately there's no standard among file managers to select the a certain file. Some don't support this feature at all, some require a certain command line option, some do it automatically, ... So it's not really possible to make this work automatically from FSearch with a simple checkbox in the settings.

> Another possibility is to allow the specification of a program to perform the directory-opening. For example, if I call dolphin [filename], then the file directory is opened and the file is also selected therein.

That's already possible, but the setting is not yet exposed in the graphical preferences. To enable it do the following:

1. Close FSearch
2. Open `~/.config/fsearch/fsearch.conf` with a text editor of your choice
3. Add the following at the end of the file:
```
    [Applications]
    folder_open_cmd=dolphin --select {path_full}
```

Now when you double click on the path column the files should get selected in Dolphin.

**Reply by pglpm** [reply]
Thank you so much Christian, this is a life-changer!
Changing a config file is simple, so this is great. Maybe this was in the Help documents – apologies if I missed it in case!

**Reply by gala8y** [reply]
Hi, is this possible with Nemo? I searched a bit for a similar syntax, but no go.

**Reply by cboxdoerfer** [reply]
@gala8y No, I don't think Nemo supports that feature. Last time I tried it, it did select files when you passed a file as an argument, e.g.  `nemo /home/test/test.txt`, but this doesn't work when you pass a folder, because it opens it instead of selecting it.

So you might want to open a feature request for that at https://github.com/linuxmint/nemo/issues, if this hasn't already been requested.

**Reply by xyTo** [reply]
@cboxdoerfer 
thanks for the detailed explanation above.
.config modification works with FSearch 0.2.2 too - only with ctrl+enter (or context menu) not the double click on path though.

exposing this (at least for the supported file managers) would surely be appreciated.

**Reply by Jetcser** [reply]
I have been using it for more than a year, which is very inconvenient when locating a file. I didn't find this setting until today, lol. I even tried to modify the source code to achieve this function before. And now you only need to modify the configuration file. Thank you very much!

**Reply by atrugiel69** [reply]
Ignore, it works with Nemo. I was just not double clicking on the path portion of the search results.

**Reply by rujiel** [reply]
Hi atrugiel69, what did you put to make this work in nemo?

**Reply by atrugiel69** [reply]
> Hi atrugiel69, what did you put to make this work in nemo?


I closed FSearch
I opened ~/.config/fsearch/fsearch.conf with a text editor
I added the following at the end of the file:

    [Applications]
    folder_open_cmd=nemo {path_full}

Then restarted FSearch.

Now if I double click on the path portion of the results, the containing folder of a file is opened with the file selected.

EDIT: Unless you wanted it to select a folder when double clicking on its path. That I don't think it's possible, as it opens the folder instead.

**Comment by rujiel** [comment]
Thank you, that works. Previously I was putting the entry in the middle of
the file, and it was disappearing. Now that I placed it at the end, it's
retained although it moves lines in the file. Thanks again.

**Reply by atrugiel69** [reply]
Which is rather funny, because I used the text previously posted as a template to explain what I did but, in my specific case, that entry is in the middle of the file... :-D

Important thing is it works for you, though. :-)
